### PR TITLE
Add command to change password of logged in user

### PR DESCRIPTION
**Pull request by xenrox**
I used this to learn how to read XEPs and how stanzas work.
I followed https://xmpp.org/extensions/xep-0077.html#usecases-changepw

Open questions:
- How to name command/functions
- Should we confirm the password (ask user to enter it twice)
- Check connection encryption?

### CODE REVIEW

**Comment by jubalh** [comment]
>    How to name command/functions

`changepassword` looks fine to me.

>   Should we confirm the password (ask user to enter it twice)

Might be handy!

>  Check connection encryption?

 I wouldn't bother. User knows his settings and is logged in already.

**Comment by xenrox** [comment]
Confirmation is in and CI is hopefully fixed as well (really miss SSHing into the build).

File: src/command/cmd_funcs.c [hunk 1]
```diff
@@ -9298,3 +9298,30 @@ cmd_mam(ProfWin* window, const char* const command, gchar** args)
...
+
+    if (g_strcmp0(passwd, confirm_passwd) != 0) {
+        cons_show("Aborted! The new password and the confirmed password do not match.");
+        return TRUE;
+    }
+
+    iq_register_change_password(user, passwd);
+
+    free(user);
+    free(passwd);
```
Comment (jubalh) [inline src/command/cmd_funcs.c:9324]: Freein confirm_passwd is missing.

**Review by jubalh — CHANGES_REQUESTED** [review]

File: src/command/cmd_funcs.c (hunk 1 above)
Comment (xenrox) [reply src/command/cmd_funcs.c:9324]: Now freed.

**Comment by jubalh** [comment]
Two more problems:
You don't free in the case that passwords don't match.
You should turn it around and only send the IQ if they match. Then you dont need the return and can use the free each time.

Also a message upon success would be good :)

**Review by jubalh — APPROVED** [review]

**Comment by jubalh** [comment]
Local `make check` passed. Code seems good. I thought maybe we should not allow empty passwords.. but you never know :-)

**Comment by jubalh** [comment]
Thanks @xenrox !

**Pull request merged**